Storage/tests/Unit/Connection/RestTest.php

```diff
@@ -466,6 +466,54 @@ public function validationMethod()
         ];
     }
 
+    /**
+     * @dataProvider retryFunctionReturnValues
+     */
+    public function testRetryFunction(
```

Comment:
Can we add a unit test for `isPreconditionSupplied` method?

Reply:
`isPreConditionSupplied` is a private function and it is unittested internally with our current unit tests itself. Should we add this?

Reply:
Done